### PR TITLE
change various expressions to use pname and version

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
there seems to be no reason for mercurial_4 to have the src like this

```
, mercurialSrc ? fetchurl rec {
    meta.name = "mercurial-${meta.version}";
    meta.version = "4.9.1";
    url = "https://mercurial-scm.org/release/${meta.name}.tar.gz";
    sha256 = "0iybbkd9add066729zg01kwz5hhc1s6lhp9rrnsmzq6ihyxj3p8v";
  }
```

File: pkgs/misc/cups/drivers/mfcl2700dncupswrapper/default.nix [hunk 1]
```diff
@@ -1,10 +1,12 @@
 { coreutils, dpkg, fetchurl, gnugrep, gnused, makeWrapper, mfcl2700dnlpr, perl, lib, stdenv }:
 
 stdenv.mkDerivation rec {
-  name = "mfcl2700dncupswrapper-${meta.version}";
+
+  pname = "mfcl2700dncupswrapper";
```
Comment (SuperSandro2000) [inline pkgs/misc/cups/drivers/mfcl2700dncupswrapper/default.nix:5]:
```suggestion
  pname = "mfcl2700dncupswrapper";
```

File: pkgs/tools/misc/lnav/default.nix [hunk 2]
```diff
@@ -3,14 +3,14 @@
 
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/lnav/default.nix:5]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/tools/misc/rolespec/default.nix [hunk 3]
```diff
@@ -2,14 +2,14 @@
 
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/rolespec/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/tools/misc/editorconfig-core-c/default.nix [hunk 4]
```diff
@@ -1,14 +1,15 @@
 { lib, stdenv, fetchgit, cmake, pcre, doxygen }:
 
 stdenv.mkDerivation rec {
-  name = "editorconfig-core-c-${meta.version}";
+
+  pname = "editorconfig-core-c";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/editorconfig-core-c/default.nix:5]:
```suggestion
  pname = "editorconfig-core-c";
```

File: pkgs/development/tools/literate-programming/funnelweb/default.nix [hunk 5]
```diff
@@ -2,7 +2,9 @@
 
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/literate-programming/funnelweb/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128843` at 9ceb7f39 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/409369b77ebb127b8f88f6b942cf5606">libsForQt514.plasma-sdk</a></li>
  </ul>
</details>
<details>
  <summary>80 packages skipped due to time constraints:</summary>
  <ul>
    <li>amarok (amarok-kf5)</li>
    <li>kbibtex</li>
    <li>kdev-php</li>
    <li>kdev-python</li>
    <li>kdevelop</li>
    <li>kdevelop-unwrapped</li>
    <li>kexi</li>
    <li>kile</li>
    <li>libsForQt5.akregator (libsForQt515.akregator ,plasma5Packages.akregator)</li>
    <li>libsForQt5.dolphin (libsForQt515.dolphin ,plasma5Packages.dolphin)</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>27 packages built successfully:</summary>
  <ul>
    <li>editorconfig-core-c</li>
    <li>funnelweb</li>
    <li>libsForQt5.dragon (libsForQt515.dragon ,plasma5Packages.dragon)</li>
    <li>libsForQt5.kde-cli-tools (libsForQt515.kde-cli-tools ,plasma5Packages.kde-cli-tools)</li>
    <li>libsForQt5.khotkeys (libsForQt515.khotkeys ,plasma5Packages.khotkeys)</li>
    <li>libsForQt5.kmail-account-wizard (libsForQt515.kmail-account-wizard ,plasma5Packages.kmail-account-wizard)</li>
    <li>libsForQt5.kmenuedit (libsForQt515.kmenuedit ,plasma5Packages.kmenuedit)</li>
    <li>libsForQt5.kompare (libsForQt515.kompare ,plasma5Packages.kompare)</li>
    <li>libsForQt5.krohnkite (libsForQt515.krohnkite ,plasma5Packages.krohnkite)</li>
    <li>libsForQt5.ktexteditor (libsForQt515.ktexteditor ,plasma5Packages.ktexteditor)</li>
    <li>libsForQt5.kwin-dynamic-workspaces (libsForQt515.kwin-dynamic-workspaces ,plasma5Packages.kwin-dynamic-workspaces)</li>
    <li>libsForQt5.kwin-tiling (libsForQt515.kwin-tiling ,plasma5Packages.kwin-tiling)</li>
    <li>libsForQt5.parachute (libsForQt515.parachute ,plasma5Packages.parachute)</li>
    <li>libsForQt5.pim-data-exporter (libsForQt515.pim-data-exporter ,plasma5Packages.pim-data-exporter)</li>
    <li>libsForQt5.plasma-browser-integration (libsForQt515.plasma-browser-integration ,plasma5Packages.plasma-browser-integration)</li>
    <li>libsForQt5.plasma-disks (libsForQt515.plasma-disks ,plasma5Packages.plasma-disks)</li>
    <li>libsForQt5.plasma-sdk (libsForQt515.plasma-sdk ,plasma5Packages.plasma-sdk)</li>
    <li>libsForQt5.plasma-workspace (libsForQt515.plasma-workspace ,plasma5Packages.plasma-workspace)</li>
    <li>libsForQt5.systemsettings (libsForQt515.systemsettings ,plasma5Packages.systemsettings)</li>
    <li>libsForQt514.ktexteditor</li>
    <li>lnav</li>
    <li>mfcl2700dncupswrapper</li>
    <li>pantheon.elementary-code</li>
    <li>rolespec</li>
    <li>sec</li>
    <li>tortoisehg</li>
    <li>yank</li>
  </ul>
</details>
<details>
  <summary>6 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/tools/literate-programming/funnelweb/default.nix:27:5:
```
   |
27 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/literate-programming/funnelweb/default.nix:18:3:
```
   |
18 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/misc/rolespec/default.nix:25:3:
```
   |
25 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/literate-programming/funnelweb/default.nix:13:3:
```
   |
13 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/misc/cups/drivers/mfcl2700dncupswrapper/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/rolespec/default.nix:43:5:
```
   |
43 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/applications/version-management/mercurial/4.9.nix [hunk 6]
```diff
@@ -1,21 +1,19 @@
...
-  }
 }:
 
 let
   inherit (python2Packages) docutils hg-git dulwich python;
 
-in python2Packages.buildPythonApplication {
+in python2Packages.buildPythonApplication rec {
+  pname = "mercurial";
+  version = "4.9.1";
```
Comment (AndersonTorres) [inline pkgs/applications/version-management/mercurial/4.9.nix:11]: According to official page, Mercurial 5.2 accepts Python 3.

Just a reminder.

Comment (Artturin) [reply pkgs/applications/version-management/mercurial/4.9.nix:11]: https://github.com/NixOS/nixpkgs/pull/76903

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
@Artturin please resolve the merge conflict.

**Comment by Artturin** [comment]
yep rebase is better than merge

**Pull request merged**